src/test/java/hudson/security/LDAPSecurityRealm_Test.java

```diff
@@ -68,4 +70,18 @@ private void check() {
         assertEquals("mAAN", sr.getMailAddressAttributeName());
     }
 
+    @Issue("JENKINS-8152")
+    @WithoutJenkins
+    @Test public void providerUrl() throws Exception {
+        assertEquals("ldap://example.com/", LDAPSecurityRealm.toProviderUrl("example.com", null));
+        assertEquals("ldap://example.com/", LDAPSecurityRealm.toProviderUrl("example.com", ""));
+        assertEquals("ldap://example.com/", LDAPSecurityRealm.toProviderUrl("example.com", "   "));
```

Comment:
rootDN is trimmed in the constrictor - so this is testing behaviour that should never occur.